src/fsharp/FSharp.Compiler.Service/FSharp.Compiler.Service.fsproj

```diff
@@ -351,6 +384,13 @@
     <Compile Include="..\ReferenceResolver.fs">
       <Link>ReferenceResolution\ReferenceResolver.fs</Link>
     </Compile>
+    <!-- A legacy resolver used to help with scripting diagnostics in the Visual Studio tools -->
+    <Compile Include="..\LegacyMSBuildReferenceResolver.fsi">
+      <Link>ReferenceResolution/LegacyMSBuildReferenceResolver.fsi</Link>
+    </Compile>
+    <Compile Include="..\LegacyMSBuildReferenceResolver.fs" Condition="'$(MonoPackaging)' != 'true'">
```

Comment:
So is `LegacyMSBuildReferenceResolver` in FSharp.Compiler.Service API surface area or not?   There should be nothing conditional on MonoPackaging anymore.

I think it should not be in this DLL (because AFAIK is the thing that induces extra dependencies on MSBuild) and should instead in a separate `FSharp.Compiler.Private.LegacyMSBuildReferenceResolver.dll`?

Reply:
@dsyme, those  msbuild dependencies have been in FCS for quite a while already.